### ISSUE TITLE
Use Mockaco in an ASP.NET Core app

### ISSUE THREAD

**Issue by gomete** [natenho/Mockaco#88]
Hi there,

I have been checking Mockaco to mock HTTP requests and I love how it works and how templating is conceived but it would be nice to extend it using the ASP.NET capabilities. Have you ever considered dividing Mockaco in the current dotnet tool and a package that can be used in an ASP.NET Core app in a similar way [YARP did it](https://microsoft.github.io/reverse-proxy/articles/getting-started.html#update-startup)?

Some of the advantages:

- Users will be able to add middlewares to the mocking pipeline (e.g, if they want to log the request or whatever).
- Users can add other controllers or endpoints can be added and manage the priority.
- Users can use all the packages that ASP.NET Core also has like OpenTelemetry, prometheus-net.AspNetCore, etc.
- Users can use their custom authentication/authorization.

I did a quick probe of concept and it is almost prepared (as it is using middlewares and the built-in dependency injection) to use it this way and it can be reused in the dotnet tool. **If you are interested I can do the changes in a fork and submit a PR to discuss it**.

So it could be used (in a similar way to YARP), registering the services and configuration:

```csharp
    public static IServiceCollection AddCustomMockaco(this IServiceCollection services) =>
		services
	        .AddMockaco()
    		.LoadFromFolder("Mocks")
			.Services;
```

And then mapping the mock endpoints with the `IApplicationBuilder`:

**Without any customization in the pipeline**:

```csharp
    public static IEndpointConventionBuilder MapCustomMockaco(this IEndpointRouteBuilder app) =>
        app.MapMockaco();
```

**Customizing the pipeline**:

```csharp
    public static IEndpointConventionBuilder MapCustomMockaco(this IEndpointRouteBuilder app) =>
        app.MapMockaco(mockingPipeline =>
        {
            mockingPipeline.UseMiddleware<LoggingMiddleware>();
            mockingPipeline.UseMiddleware<AuditingMiddleware>();
        });
```

**Comment by natenho** [comment]
Hello @gomete, it sounds great! The idea is to let Mockaco be injected and extended by another applications, right?
I would appreciate if you could share your PoC via PR, and we can discuss it further!

It somehow talks to #66 as it is some kind of extension.

**Comment by gomete** [comment]
Yes, that's the idea @natenho, let me share it today with you via PR.

**Comment by gomete** [comment]
Hi @natenho, PR created #89

**Comment by natenho** [comment]
Hi @gomete, I've finally could check out and merge your PR! 
We can improve the feature as you suggested in the issue, but I think your current implementation is a good starting point with no breaking change, so I've made it public "as-is".

I had to do some adjustments and to tackle with some conflicts but it seems everything is OK now! The new nuget package is already available at https://www.nuget.org/packages/Mockaco.AspNetCore

Thanks again for the feature and for your patience.
